src/ORM/Search/SearchContext.php

```diff
@@ -176,40 +205,117 @@ public function getQuery($searchParams, $sort = false, $limit = false, $existing
+                    // to provide some unexpected field name, where the below would result in a DatabaseException.
+                    if ((!isset($spec['primary']) || $spec['primary'])
+                        && ($formFields->fieldByName($formFieldName) || $formFields->dataFieldByName($formFieldName))
+                        && $filter !== null
+                    ) {
+                        $filter->setModel($modelClass);
+                        $filter->setValue($searchPhrase);
+                        $context->applyFilter($filter, $subGroup, $spec);
+                    }
+                }
+            };
+
+            // If necessary, split search phrase into terms, then search across fields.
+            if (Config::inst()->get($modelClass, 'primary_search_split_terms')) {
+                if (is_array($searchPhrase)) {
+                    // Allow matches from ANY query in the array (i.e. return $obj where query1 matches OR query2 matches)
+                    $dataQuery = $dataQuery->disjunctiveGroup();
+                    foreach ($searchPhrase as $phrase) {
+                        // where ((field1 like lorem OR field2 like lorem) AND (field1 like ipsum OR field2 like ipsum))
+                        $primarySubGroup = $dataQuery->conjunctiveGroup();
+                        foreach (explode(' ', $phrase) as $searchTerm) {
+                            $searchAcrossFields($searchTerm, $primarySubGroup->disjunctiveGroup());
+                        }
+                    }
+                } else {
+                    // where ((field1 like lorem OR field2 like lorem) AND (field1 like ipsum OR field2 like ipsum))
+                    $primarySubGroup = $dataQuery->conjunctiveGroup();
+                    foreach (explode(' ', $searchPhrase) as $searchTerm) {
+                        $searchAcrossFields($searchTerm, $primarySubGroup->disjunctiveGroup());
                     }
                 }
+            } else {
+                // where (field1 like lorem ipsum OR field2 like lorem ipsum)
+                $searchAcrossFields($searchPhrase, $dataQuery->disjunctiveGroup());
             }
+        });
+    }
+
+    /**
+     * Get the search filter for the given fieldname when searched from the primary search field.
+     */
+    public function getPrimarySearchFilter(string $modelClass, string $fieldName): ?SearchFilter
+    {
+        if ($filterClass = Config::inst()->get($modelClass, 'primary_search_field_filter')) {
+            return Injector::inst()->create($filterClass, $fieldName);
         }
+        return $this->getFilter($fieldName);
+    }
 
+    /**
+     * Search against a single field
+     *
+     * @param string|array $searchPhrase
+     */
+    protected function individualFieldSearch(DataList $query, array $searchableFields, string $searchField, $searchPhrase): DataList
+    {
+        if ($filter = $this->getFilter($searchField)) {
```

Comment:
```suggestion
        $filter = $this->getFilter($searchField);
        if (!$filter) {
            return $query;
        }
```
I've never liked the assignment and conditional syntax, I find it hard to read.  Can do an early return here as well to remove the indent below
I find putting on a one-liner much harder to read

Reply:
Done